### DISCUSSION TITLE
I have a few endpoints needed for authentication, but none of them would be in the openapi file. Any ideas on how I can integrate with my generated postman collection?

### DISCUSSION THREAD

**Discussion by whorst** [apideck-libraries/portman#243] [General]
I have about 3 google endpoints my team has to use for authentication to get into our application. None of those 3 endpoints would necessarily need to be anywhere in the file where all of our OpenAPI documentation lives (this is the file that portman would ingest), but capturing the response values from the google auth endpoints (mostly authentication tokens) at the collection variable level and using them in subsequent requests to our endpoints would make life extremely easy.

So, any ideas on how I could make this work? Ideally, I'd have the three endpoints exist in the same portman generated postman collection that I could easily run.

Let me know

**Comment by whorst** [comment]
I found some more cool open source tech to help me. I ended up making a separate postman collection for all of my auth stuff. So I now have two collection, the API collection and the Auth API collection. I combined both of those into one with 
https://www.npmjs.com/package/postman-combine-collections

Pretty nifty

**Comment by thim81** [comment]
The Postman merge is an approach that could work.

I'm not sure what you try to achieve.
Personally I needed an Oauth flow, which requests an Access token before executing any Postman requests.
The build-in Postman auth flows are confusing and difficult to automate when using OpenAPI conversions, so I have setup 2 approaches:

- Using pre-request script and storing the Oauth Token in my Postman variables
- Using a nodeJS script that requests the Oauth Token and generates a Postman variables files, which I can use in Postman / Newman

[Example for the "Pre-request script"](https://gist.github.com/thim81/a1e218272fbcf934bcbc2d19cab70fef)
[Example for the "nodeJS script"](https://gist.github.com/thim81/767d7627a45aa66978e9368e0bfc383e)

**Reply by savage-alex** [reply]
+1 for the pre-request scripts. You can inject these into your portman config too! We are using authcode which is tricky to automate so have to seed the environment with a refresh token then it will retrieve access tokens and keep them alive as needed.
